yolox/core/trainer.py

```diff
@@ -336,16 +339,17 @@ def evaluate_and_save_model(self):
                 self.wandb_logger.log_metrics({
                     "val/COCOAP50": ap50,
                     "val/COCOAP50_95": ap50_95,
-                    "epoch": self.epoch + 1,
+                    "train/epoch": self.epoch + 1,
                 })
+                self.wandb_logger.log_images(predictions)
             logger.info("\n" + summary)
         synchronize()
 
-        self.save_ckpt("last_epoch", update_best_ckpt)
+        self.save_ckpt("last_epoch", update_best_ckpt, ap=ap50_95)
         if self.save_history_ckpt:
-            self.save_ckpt(f"epoch_{self.epoch + 1}")
+            self.save_ckpt(f"epoch_{self.epoch + 1}", ap=ap50_95)
 
-    def save_ckpt(self, ckpt_name, update_best_ckpt=False):
+    def save_ckpt(self, ckpt_name, update_best_ckpt=False, ap=None):
```

Comment:
The reason for adding `ap` as an argument was to be able to log it as metadata along with the model parameters. Making `current_ap` an attribute would remove the requirement for changing the function signature so that would be a better case. Will update this.